### DISCUSSION TITLE
Blackout, Menu Développeur, Apparts

### DISCUSSION THREAD

**Discussion by Maskowh** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#96] [General]
Bonjour,

Vu que je laisse pas mal tourner mon serveur dev, je me retrouve facilement en blackout UPW

Je connais pas du tout le mode de fonctionnement d'UPW, est ce qu'il y a moyen de remettre le courant avec les onduleurs?
Ou il faut se créer des bornes civiles pour ça?

Bon sinon je peux aller directement modifier ce qu'il faut en base, mais je me demandais si y avais une méthode moins bourrin.

Et au passage, vous êtes ouverts aux contributions pour le menu d'Outils Développeurs? Par exemple, Déclencher un blackout, Remplir/Vider toutes les stations essences, Remplir/Vider toutes les bornes UPW, 

Enfin, est-il possible d'avoir une explication sur le placement d'appartements/garages/bornes UPW?

Merci!

**Comment by DreamXZE** [comment]
Hello, oui il y a une method moins bourrine!
Dans ton `env.cfg` tu peux modifier:
- `soz_upw_energy_per_cell_fossil`/`soz_upw_energy_per_cell_hydro`/`soz_upw_energy_per_cell_wind` et mettre des valeur hautes et recharger plus facilement
- `soz_upw_waste_per_harvest` à 0 pour ne pas créer de déchet pret du barage quand tu récupère des cellule hydro
- `soz_upw_consumption_energy_per_tick`/`soz_upw_consumption_energy_job_per_tick` à 0 pour que l'energie ne descende pas du tout

Pour le reste, je laisse quelqu'un d'autre te répondre :pray:

**Comment by poulpito** [comment]
Sinon
dans soz-upw/server/storage.lua    tu coupe la consumptionloop  
![image](https://user-images.githubusercontent.com/4135866/216648654-21acae54-363a-441e-9466-944c989d81f4.png)

Pour la commande de blackout elle existe déjà (en commande /blackout level de mémoire) mais n'est pas dans un menu 

Pour les placements appartement / garage  c'est avec le menu F10 de mapping  gestion des propriétés 
tu dois déclarer une maison en déclarant sa position d'arrivée dans le mapping / sortie directement en éditant ingame des zones de B-target
![image](https://user-images.githubusercontent.com/4135866/216649911-202f95d0-a3dc-4b65-87a7-a2cccac62cfe.png)
La zone est éditable avec différentes touches (ctrl / maj / caps lock) et déplacable avec les fleches  

Pour les props comme les bornes élect et poubelles  c'est toujours via F10 mais avec gestion des objets et l'entreprise qui y est attachée (et qui sera prise par le script par ex pour les bornes UPW)

**Comment by Maskowh** [comment]
Merci à tous les deux pour vos explications!
